helix-term/src/keymap.rs

```diff
@@ -468,169 +472,222 @@ impl Keymaps {
-            "end" => goto_line_end,
-
-            "w" => move_next_word_start,
-            "b" => move_prev_word_start,
-            "e" => move_next_word_end,
-
-            "W" => move_next_long_word_start,
-            "B" => move_prev_long_word_start,
-            "E" => move_next_long_word_end,
-
-            "v" => select_mode,
-            "G" => goto_line,
-            "g" => { "Goto"
-                "g" => goto_file_start,
-                "e" => goto_last_line,
-                "f" => goto_file,
-                "h" => goto_line_start,
-                "l" => goto_line_end,
-                "s" => goto_first_nonwhitespace,
-                "d" => goto_definition,
-                "y" => goto_type_definition,
-                "r" => goto_reference,
-                "i" => goto_implementation,
-                "t" => goto_window_top,
-                "c" => goto_window_center,
-                "b" => goto_window_bottom,
-                "a" => goto_last_accessed_file,
-                "m" => goto_last_modified_file,
-                "n" => goto_next_buffer,
-                "p" => goto_previous_buffer,
-                "." => goto_last_modification,
-            },
-            ":" => command_mode,
-
-            "i" => insert_mode,
-            "I" => prepend_to_line,
-            "a" => append_mode,
-            "A" => append_to_line,
-            "o" => open_below,
-            "O" => open_above,
-
-            "d" => delete_selection,
-            "A-d" => delete_selection_noyank,
-            "c" => change_selection,
-            "A-c" => change_selection_noyank,
-
-            "C" => copy_selection_on_next_line,
-            "A-C" => copy_selection_on_prev_line,
-
-
-            "s" => select_regex,
-            "A-s" => split_selection_on_newline,
-            "S" => split_selection,
-            ";" => collapse_selection,
-            "A-;" => flip_selections,
-            "A-k" | "A-up" => expand_selection,
-            "A-j" | "A-down" => shrink_selection,
-            "A-h" | "A-left" => select_prev_sibling,
-            "A-l" | "A-right" => select_next_sibling,
-
-            "%" => select_all,
-            "x" => extend_line,
-            "X" => extend_to_line_bounds,
-            // crop_to_whole_line
-
-            "m" => { "Match"
-                "m" => match_brackets,
-                "s" => surround_add,
-                "r" => surround_replace,
-                "d" => surround_delete,
-                "a" => select_textobject_around,
-                "i" => select_textobject_inner,
-            },
-            "[" => { "Left bracket"
-                "d" => goto_prev_diag,
-                "D" => goto_first_diag,
-                "f" => goto_prev_function,
-                "c" => goto_prev_class,
-                "a" => goto_prev_parameter,
-                "o" => goto_prev_comment,
-                "space" => add_newline_above,
+fn default_keymaps() -> HashMap<Mode, Keymap> {
```

Comment:
(1/2) Had to move the default keymaps into its own function, this was to allow...